rust/metaplex/program/src/processor/empty_payment_account.rs

```diff
@@ -85,11 +85,18 @@ fn calculate_owed_amount(
     winning_config_item_index: &Option<u8>,
     creator_index: &Option<u8>,
 ) -> Result<u64, ProgramError> {
-    let primary_sale_happened = auction_manager.get_primary_sale_happened(
-        metadata,
-        *winning_config_index,
-        *winning_config_item_index,
-    )?;
+    let primary_sale_happened = match safety_deposit_config_info.and_then(|a| {
```

Comment:
This is the primary change. It switches to reading primary sale happened from the metadata account to the snapshot of primary sale happened saved to the safety deposit config at the time of listing.

It is handling checking for v1 and v2 configs without introducing a new empty payment instruction.